cellpack/autopack/ingredient/single_cube.py

```diff
@@ -172,38 +188,43 @@ def collision_jitter(
             bb, center_trans, search_radius
         )  # indices of all grid points within padded distance from cube center
 
-        grid_point_vectors = (
-            numpy.take(gridPointsCoords, points_to_check, 0) - center_trans
-        )  # vectors joining center of cube with grid points
-        grid_point_distances = numpy.linalg.norm(
-            grid_point_vectors, axis=1
-        )  # distances of grid points from packing location
+        grid_point_vectors = numpy.take(gridPointsCoords, points_to_check, 0)
+
+        # signed distances of grid points from the cube surface
+        grid_point_distances = []
+        for grid_point in grid_point_vectors:
+            grid_point_distance = self.get_signed_distance(
+                center_trans,
+                grid_point,
+                rotMat,
+            )
+            grid_point_distances.append(grid_point_distance)
 
         for pti in range(len(points_to_check)):
             # pti = point index
 
             grid_point_index = points_to_check[pti]
-
-            if grid_point_index in insidePoints:
-                continue
+            signed_distance_to_cube_surface = grid_point_distances[pti]
 
             collision = (
-                grid_point_distances[pti] + current_grid_distances[grid_point_index]
-                <= self.encapsulatingRadius
+                signed_distance_to_cube_surface
+                + current_grid_distances[grid_point_index]
+                <= 0
             )
 
             if collision:
                 self.log.info("grid point already occupied %f", grid_point_index)
                 return True, {}, {}
 
-            # currently calculates distance from surface of encapsulating sphere
-            # should be updated to distance from cube surface
-            signed_distance_to_cube_surface = (
-                grid_point_distances[pti] - self.encapsulatingRadius
-            )
             if signed_distance_to_cube_surface <= 0:
                 # check if grid point lies inside the cube
-                insidePoints[grid_point_index] = signed_distance_to_cube_surface
+                if grid_point_index in insidePoints:
+                    if abs(signed_distance_to_cube_surface) < abs(
+                        insidePoints[grid_point_index]
+                    ):
+                        insidePoints[grid_point_index] = signed_distance_to_cube_surface
+                else:
+                    insidePoints[grid_point_index] = signed_distance_to_cube_surface

```

Comment:
Isn't this just 
```
if grid_point_index not in insidePoints or abs(signed_distance_to_cube_surface) < abs(
        insidePoints[grid_point_index]
    ):
        insidePoints[grid_point_index] = signed_distance_to_cube_surface
```
?

Reply:
Yup, will implement this change.

It makes the flow cleaner, thanks!